pact/contracts/pact_test.go

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contracts
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/pact-foundation/pact-go/v2/provider"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	oauth "github.com/redhat-appstudio/service-provider-integration-operator/oauth"
+)
+
+func TestContracts(t *testing.T) {
+	// Register fail handler and setup test environment (same as during unit tests)
+	RegisterFailHandler(Fail)
+	IT, ctx = oauth.StartTestEnv()
+
+	// Create and setup Pact Verifier
+	verifyRequest := createVerifier(t)
+
+	// Run pact tests
+	err := provider.NewVerifier().VerifyProvider(t, verifyRequest)
+	if err != nil {
+		t.Errorf("Error while verifying tests. \n %+v", err)
+	}
+
+	cleanUpNamespaces()
+
+	IT.Cancel()
+
+	err = IT.TestEnvironment.Stop()
+	if err != nil {
+		fmt.Println("Stopping failed")
+		fmt.Printf("%+v", err)
+		panic("Cleanup failed")
+	}
+}
+
+func createVerifier(t *testing.T) provider.VerifyRequest {
+	verifyRequest := provider.VerifyRequest{
+		Provider:        "SPI",
+		RequestTimeout:  60 * time.Second,
+		ProviderBaseURL: IT.TestEnvironment.Config.Host,
+		// Default selector should include environments, but as they are not in place yet, using just main branch
+		ConsumerVersionSelectors:   []provider.Selector{&provider.ConsumerVersionSelector{Branch: "main"}},
+		BrokerURL:                  "https://pact-broker-hac-pact-broker.apps.hac-devsandbox.5unc.p1.openshiftapps.com",
+		PublishVerificationResults: false,
+		EnablePending:              true,
+		ProviderVersion:            "local",
+		ProviderBranch:             "main",
+	}
+
+	// clean up test env before every test
+	verifyRequest.BeforeEach = func() error {
+		// workaround for https://github.com/pact-foundation/pact-go/issues/359
+		if os.Getenv("SETUP") == "true" {
+			return nil
+		}
+		cleanUpNamespaces()
+		return nil
+	}
+
+	// setup credentials and publishing
+	if os.Getenv("PR_CHECK") != "true" {
```

Comment:
Yes, so far it will run only on PR check and will not report back to Pact broker. I would like to add a post-merge job that will run the tests for the merged branch and report the results back to the Pact broker. I planned to add the job as a separate PR.